datalad_catalog/workflows.py

```diff
@@ -0,0 +1,248 @@
+from datalad.api import (
+    meta_extract,
+    meta_conduct,
+)
+from datalad.distribution.dataset import Dataset
+from datalad.support.exceptions import IncompleteResultsError
+from pathlib import Path
+import json
+from uuid import UUID
+import jq
+from datalad_catalog.utils import read_json_file
+from datalad_catalog.webcatalog import WebCatalog
+from datalad_catalog.catalog import (
+    _add_to_catalog,
+)
+
+# DETAILS FOR EXTRACTORS AND TRANSLATORS
+extractor_names_dataset = [
+    "metalad_core",
+    "metalad_studyminimeta",
+    "bids_dataset",
+    "datacite_gin",
+]
+extractor_names_file = [
+    "metalad_core",
+]
+required_files = dict(
+    metalad_core=[""],
+    metalad_studyminimeta=[".studyminimeta.yaml"],
+    bids_dataset=["participants.tsv", "dataset_description.json"],
+    datacite_gin=["datacite.yml"],
+)
+translator_map = {
+    "metalad_core": {
+        "file": "_metaladcore2catalog_file.json",
+        "dataset": "_metaladcore2catalog_dataset.json",
+    },
+    "metalad_studyminimeta": "_studyminimeta2catalog.json",
+    "bids_dataset": "_bidsdataset2catalog.json",
+    "datacite_gin": "_datacitegin2catalog.json",
+}
+# INTERNAL PATHS
+package_path = Path(__file__).resolve().parent
+schema_dir = package_path / "schema"
+
+
+class jsEncoder(json.JSONEncoder):
+    """Class to return objects as strings for correct JSON encoding"""
+
+    def default(self, obj):
+        if isinstance(obj, UUID) or isinstance(obj, Path):
+            # if the obj is uuid, we simply return the value of uuid
+            return str(obj)
+
+        return json.JSONEncoder.default(self, obj)
+
+
+def super_workflow(dataset_path, catalog: WebCatalog):
+    """Run a workflow from scratch on a dataset and all its subdatasets
+
+    The workflow includes:
+    - Recursively installing the super- and subdatasets
+    - Creating the catalog if it does not yet exists
+    - Running several steps on the super- and subdatasets:
+      - dataset- and file-level metadata extraction
+      - extracted metadata translation
+      - adding translated metadata to the catalog
+    - Setting the super-dataset of catalog
+    """
+    # Install super and subdatasets
+    ds = Dataset(dataset_path)
+    # ds.get(get_data=False, recursive=True)
+    # Create catalog
+    cat = catalog
+    if not cat.is_created():
+        cat.create()
+    # Call per-dataset workflow
+    def _dataset_workflow_inner(ds, refds, **kwargs):
+        """Internal function to allow passing"""
+        return dataset_workflow(ds, catalog=cat, **kwargs)
+
+    try:
+        for res in ds.foreach_dataset(
+            _dataset_workflow_inner,
+            recursive=True,
+            recursion_limit=1,
```

Comment:
So it does superdataset & its subdatsets, but not deeper? I'm OK with that, but maybe worth mentioning explicitly in description?

Reply:
Yes, that's correct, and agreed it would be good to mention this. A possibility would also be to allow the user to specify the recursion limit, and to build the automation around this (and other) parameters. But I think it makes more sense to leave such a refactoring for a future issue/PR.